Annotato/Annotato/WebSocket/WebSocketManager.swift

```diff
@@ -1,18 +1,28 @@
 import Foundation
+import Combine
 import AnnotatoSharedLibrary
 
 class WebSocketManager: ObservableObject {
     private(set) var socket: URLSessionWebSocketTask?
+    private var urlStringProducer: () -> String
```

Comment:
```suggestion
    private var urlStringProducer: () -> String?
```

To accommodate the case where the string could not be produced